{
    "schema_version": "1.6.0",
    "id": "OESA-2022-2055",
    "modified": "2022-11-11T00:00:00Z",
    "published": "2022-11-11T00:00:00Z",
    "withdrawn": "",
    "aliases": null,
    "related": [
        "CVE-2021-45115",
        "CVE-2021-45116",
        "CVE-2021-45452",
        "CVE-2022-22818",
        "CVE-2022-23833"
    ],
    "summary": "python-django security update",
    "details": "\r\n\r\nSecurity Fix(es):\r\n\r\nAn issue was discovered in Django 2.2 before 2.2.26, 3.2 before 3.2.11, and 4.0 before 4.0.1. UserAttributeSimilarityValidator incurred significant overhead in evaluating a submitted password that was artificially large in relation to the comparison values. In a situation where access to user registration was unrestricted, this provided a potential vector for a denial-of-service attack.(CVE-2021-45115)\r\n\r\nAn issue was discovered in Django 2.2 before 2.2.26, 3.2 before 3.2.11, and 4.0 before 4.0.1. Due to leveraging the Django Template Language\u0026apos;s variable resolution logic, the dictsort template filter was potentially vulnerable to information disclosure, or an unintended method call, if passed a suitably crafted key.(CVE-2021-45116)\r\n\r\nStorage.save in Django 2.2 before 2.2.26, 3.2 before 3.2.11, and 4.0 before 4.0.1 allows directory traversal if crafted filenames are directly passed to it.(CVE-2021-45452)\r\n\r\nThe {% debug %} template tag in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2 does not properly encode the current context. This may lead to XSS.(CVE-2022-22818)\r\n\r\nAn issue was discovered in MultiPartParser in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2. Passing certain inputs to multipart forms could result in an infinite loop when parsing files.(CVE-2022-23833)",
    "severity": null,
    "affected": [
        {
            "package": {
                "ecosystem": "openEuler-22.03-LTS",
                "name": "python-django",
                "purl": "pkg:rpm/python-django@2.2.27-2.oe2203"
            },
            "severity": null,
            "ranges": [
                {
                    "type": "ECOSYSTEM",
                    "repo": "",
                    "events": [
                        {
                            "introduced": "0",
                            "fixed": "2.2.27-2.oe2203",
                            "last_affected": "",
                            "limit": ""
                        }
                    ],
                    "database_specific": null
                }
            ],
            "versions": null,
            "ecosystem_specific": {
                "noarch": [
                    "python3-Django-2.2.27-2.oe2203.noarch.rpm",
                    "python-django-help-2.2.27-2.oe2203.noarch.rpm"
                ],
                "src": [
                    "python-django-2.2.27-2.oe2203.src.rpm"
                ]
            },
            "database_specific": null
        }
    ],
    "references": [
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-45115"
        },
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-45116"
        },
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-45452"
        },
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-22818"
        },
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-23833"
        }
    ],
    "credits": null,
    "database_specific": {
        "severity": "High"
    }
}